libs/forms/core/src/control-config.types.ts

```diff
@@ -1,32 +1,43 @@
-import { AbstractControlOptions } from '@angular/forms';
 import { Observable } from 'rxjs';
 import { DynControlFactoryParams, DynControlParams } from './control-params.types';
+import { DynControlTriggers } from './control-validation.types';
 import { DynControlType } from './control.types';
 
-// single dynamic control config
+export type DynConfigPrimitive = any; // FIXME omit functions?
+
+// a plain/serializable value
+export type DynConfigArgs = DynConfigPrimitive | DynConfigPrimitive[] | null;
+
+// a given id to the functions
+export type DynConfigId = string;
+
+// a collection of ids with arguments to be used
+export type DynConfigCollection = { [id: string]: DynConfigArgs }
+                                | Array<DynConfigId | [DynConfigId, DynConfigArgs]>;
```

Comment:
@Yumitoya8569 I ended up supporting 3 notations for `options: { validators }`:

1. `{ min: 3 }`
2. `[ 'required' ]`
3. `[ 'requiredTrue', ['minLength', 3]`